compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpResourceValidator.java

```diff
@@ -449,6 +455,23 @@ public static void extractInputParamTypeAndValidate(SyntaxNodeAnalysisContext ct
                 }
             }
         }
+        if (resourceMethodOptional.isPresent() && !payloadAnnotationPresent) {
+            enableAddPayloadParamCodeAction(ctx, member.location(), resourceMethodOptional.get());
+        }
+        if (!headerAnnotationPresent) {
+            enableAddHeaderParamCodeAction(ctx, member.location());
+        }
+    }
+
+    private static void enableAddPayloadParamCodeAction(SyntaxNodeAnalysisContext ctx, Location location,
+                                                       String methodName) {
```

Comment:
```suggestion
                                                        String methodName) {
```

Reply:
Fixed in [`239e4bd`](https://github.com/ballerina-platform/module-ballerina-http/pull/880/commits/239e4bdc40cf610cabdae97329c50c88c8a30f59)